app/controllers/api/orders_controller.rb

```diff
@@ -0,0 +1,65 @@
+class Api::OrdersController < Api::ApplicationController
+  include OrdersHelper
+  before_action :authenticate_user!, only: [:add_to_cart, :view_cart, :convert_cart_to_order]
+
+  def add_to_cart
+    product = Product.find(params[:product_id])
+    order = Order.find_or_create_by(user_id: current_user.id, isCart: true) do |new_order|
+      new_order.status = 0
+    end
+
+    begin
+      order_item = OrderItem.find_by(order_id: order.id, product_id: product.id)
+      if order_item.nil?
+        OrderItem.create(order_id: order.id, product_id: product.id, quantity: 1)
+      else
+        order_item.quantity += 1
+        order_item.save
+      end
+    rescue StandardError => e
+
+      render json: { error: "Error creating OrderItem: #{e.message}" }, status: :unprocessable_entity
+    end
+
+    render json: order, serializer: OrderSerializer, status: :ok
+  end
+
+  def view_cart
+    @order = Order.find_by(user_id: current_user.id, isCart: true)
```

Comment:
you did `render json: {} if condition` below, why not here as well? you can do `render json: { error: 'Cart is empty' }, status: :unprocessable_entity if @order.nil?` instead of `if order.nil? else`